pkg/values/value.go

```diff
@@ -278,10 +278,36 @@ func unwrapTo[T any](underlying T, to any) error {
 		rTo := reflect.ValueOf(to)
 		rUnderlying := reflect.ValueOf(underlying)
 		underlyingPtr := reflect.PointerTo(rUnderlying.Type())
-		if rTo.Kind() != reflect.Pointer || !rTo.CanConvert(underlyingPtr) {
+		if rTo.Kind() != reflect.Pointer {
 			return fmt.Errorf("cannot unwrap to value of type: %T", to)
 		}
-		reflect.Indirect(rTo.Convert(underlyingPtr)).Set(rUnderlying)
+
+		if rTo.CanConvert(underlyingPtr) {
+			reflect.Indirect(rTo.Convert(underlyingPtr)).Set(rUnderlying)
+			return nil
+		}
+
+		rToVal := reflect.Indirect(rTo)
+		if rToVal.Kind() == reflect.Slice && rUnderlying.Kind() == reflect.Slice {
+			for i := 0; i < rUnderlying.Len(); i++ {
+				el := rUnderlying.Index(i)
+				toEl := reflect.New(rToVal.Type().Elem())
+				err := unwrapTo(el.Interface(), toEl.Interface())
```

Comment:
Aha! Nice trick to recursively call unwrapTo and handle. In the base case where unwrapTo is called for each element, how is the following call to Set not throwing the same error:

```
reflect.Indirect(rTo.Convert(underlyingPtr)).Set(rUnderlying)
```

Error: 
```
panic: reflect: reflect.Value.Set using unaddressable value [recovered]
panic: reflect: reflect.Value.Set using unaddressable value
```

Reply:
Hey! Yeah, we avoid that problem by creating the pointer on line 294

Reply:
Ah ok. So thats what we were missing yesterday! Thanks!